.github/copilot-setup-steps.yml

```diff
@@ -0,0 +1,205 @@
+      npm install -g @azure/static-web-apps-cli@latest
+      
+  # Install frontend dependencies
+  - name: Install frontend dependencies
+    run: |
+      cd client
+      npm ci
+      
+  # Install backend dependencies
+  - name: Restore .NET packages
+    run: |
+      cd server
+      dotnet restore
+      dotnet tool restore
+      
+  # Install development tools
+  - name: Install development tools
+    run: |
+      # Language servers for MCP
+      npm install -g typescript-language-server
+      npm install -g vscode-langservers-extracted
+      npm install -g @angular/language-server
+      
+      # Code quality tools
+      npm install -g cspell
+      npm install -g markdownlint-cli
+      
+      # Git tools
+      sudo apt-get install -y git-extras
+      
+  # Configure Git for better AI assistance
+  - name: Configure Git
+    run: |
+      git config --global init.defaultBranch main
+      git config --global pull.rebase false
+      git config --global core.autocrlf input
+      
+  # Verify installations
+  - name: Verify installations
+    run: |
+      echo "Node.js version: $(node --version)"
+      echo "npm version: $(npm --version)"
+      echo "Angular CLI version: $(ng version --skip-git)"
+      echo ".NET version: $(dotnet --version)"
+      echo "Azure Functions Core Tools version: $(func --version)"
+      echo "TypeScript version: $(tsc --version)"
+      
+  # Pre-build projects to cache dependencies
+  - name: Pre-build projects
+    run: |
+      # Build frontend
+      cd client
+      npm run build --if-present
+      
+      # Build backend
+      cd ../server
+      dotnet build --no-restore
+      
+  # Create useful aliases and scripts
+  - name: Setup development aliases
+    run: |
+      cat >> ~/.bashrc << 'EOF'
+      
+      # Stock Charts Development Aliases
+      alias sc-client='cd /workspaces/stock-charts/client'
+      alias sc-server='cd /workspaces/stock-charts/server'
+      alias sc-start-client='cd /workspaces/stock-charts/client && npm start'
+      alias sc-build-client='cd /workspaces/stock-charts/client && npm run build'
+      alias sc-test-client='cd /workspaces/stock-charts/client && npm test'
+      alias sc-start-server='cd /workspaces/stock-charts/server && func start'
+      alias sc-build-server='cd /workspaces/stock-charts/server && dotnet build'
+      alias sc-test-server='cd /workspaces/stock-charts/server && dotnet test'
+      alias sc-logs='cd /workspaces/stock-charts && git log --oneline -10'
+      alias sc-status='cd /workspaces/stock-charts && git status'
+      
+      EOF
+      
+  # Setup VS Code settings for the workspace
+  - name: Configure VS Code workspace
+    run: |
+      mkdir -p /workspaces/stock-charts/.vscode
+      
+      # Ensure workspace settings are properly configured
+      cat > /workspaces/stock-charts/.vscode/tasks.json << 'EOF'
+      {
+          "version": "2.0.0",
+          "tasks": [
```

Comment:
are these also in our `.vscode/tasks.json` file?  if not add as appropriate and ensure these aren’t redundant.